packages/eslint-plugin/src/rules/ban-context-export/index.js

```diff
@@ -45,11 +44,7 @@ module.exports = createRule({
     },
   },
   create(context) {
-    const rawOptions = /** @type {Options[]} */ (context.options);
-    const { exclude = [] } = rawOptions.length ? rawOptions[0] : defaultOptions;
-    const { program, esTreeNodeToTSNodeMap } = ESLintUtils.getParserServices(context);
-    /** @type {import("typescript").TypeChecker | undefined} */
-    let typeChecker;
+    const [{ exclude = /** @type string[]*/ ([]) } = {}] = context.options;
```

Comment:
`defaultOptions` resolution is still a bit of a mystery, I wasn't able to find any docs how it actually works. any ideas where to look @ling1726 ?